File: AED/laravel/MiniDriveConDAO/app/Models/Matricula.php
<?php
    namespace App\Models;

    use App\Models\Alumno;

    class Matricula{
        private $idmatricula;
        private Alumno $alumno;
        private $year;
        private $asignaturas = [];
        
        public function getIdmatricula(){
            return $this->idmatricula;
        }

        public function setIdmatricula($id){
            $this->idmatricula = $id;
        }

        public function getAlumno(){
            return $this->alumno;
        }

        public function setAlumno($a){
            $this->alumno = $a;
        }

        public function getYear(){
            return $this->year;
        }

        public function setYear($f){
            $this->year = $f;
        }

        public function getAsignaturas(){
            return $this->asignaturas;
        }

        public function setAsignaturas($arr){
            $this->asignaturas = $arr;
        }

        public function __toString(){
            $asinaturasStr = "";
            foreach ($this->asignaturas as $key => $value) {
                $asinaturasStr .= $value->getNombre() . "<br/>";
            }

            return "Alumno: " .
            $this->alumno->getnombre() . " " . $this->alumno->getApellidos() .
            " Id: " . $this->getIdmatricula() . 
            " Year:" . $this->getYear() . 
            " Asignaturas: " . $asinaturasStr;
        }
    }
?>

File: AED/java/Spring/matriculas/src/main/java/es/iespto/agl/matriculas/entity/Matricula.java
package es.iespto.agl.matriculas.entity;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.NamedQuery;
import jakarta.persistence.Table;


/**
 * The persistent class for the matriculas database table.
 * 
 */
@Entity
@Table(name="matriculas")
@NamedQuery(name="Matricula.findAll", query="SELECT m FROM Matricula m")
public class Matricula implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(unique=true, nullable=false)
	private Integer id;

	@Column(name="[year]")
	private Integer year;

	@ManyToOne
	@JoinColumn(name="dni")
	@JsonIgnore
	private Alumno alumno;

	@ManyToMany(fetch= FetchType.LAZY)
	@JoinTable(
		name="asignatura_matricula"
		, joinColumns={
			@JoinColumn(name="idmatricula")
			}
		, inverseJoinColumns={
			@JoinColumn(name="idasignatura")
			}
		)	
	private List<Asignatura> asignaturas;

	public Matricula() {
	}

	public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Integer getYear() {
		return this.year;
	}

	public void setYear(Integer year) {
		this.year = year;
	}

	public Alumno getAlumno() {
		return this.alumno;
	}

	public void setAlumno(Alumno alumno) {
		this.alumno = alumno;
	}

	public List<Asignatura> getAsignaturas() {
		return this.asignaturas;
	}

	public void setAsignaturas(List<Asignatura> asignaturas) {
		this.asignaturas = asignaturas;
	}

}